### PR TITLE
Support rewriting OPTIMIZE and ALTER on distributed table to local table with on cluster.

### PR DESCRIPTION
**Pull request by kindred77**
Rewrite OPTIMIZE and ALTER on distributed table to corresponding local table with ON CLUSTER when enable_ddl_optimize_rewrite and enable_ddl_alter_rewrite enabled.

### Changelog category (leave one):
- Improvement

### Documentation entry for user-facing changes

Somtimes, distributed tables are used as a User Interface, it's not a good idea for user to specify the corresponding local table and cluster when they want to change data with DDL. DDL to distributed table would be rewriten to corresponding local table with ON CLUSTER option when options enabled, OPTIMIZE and ALTER are supported currently.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Need to sign a CLA.
Need to add a test.

File: src/Storages/Distributed/DistributedSettings.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ class ASTStorage;
     M(Milliseconds, background_insert_sleep_time_ms, 0, "Default - distributed_background_insert_sleep_time_ms", 0) ALIAS(monitor_sleep_time_ms) \
     M(Milliseconds, background_insert_max_sleep_time_ms, 0, "Default - distributed_background_insert_max_sleep_time_ms", 0) ALIAS(monitor_max_sleep_time_ms) \
     M(Bool, flush_on_detach, true, "Flush data to remote nodes on DETACH/DROP/server shutdown", 0) \
+    /* ddl rewrite to on cluster settings */ \
+    M(Bool, enable_ddl_optimize_rewrite, false, "OPTIMIZE will be rewrited to optimize on corresponding cluster.", 0) \
```
Comment (alexey-milovidov) [inline src/Storages/Distributed/DistributedSettings.h:33]: rewrited -> rewritten

File: src/Storages/StorageDistributed.h [hunk 2]
```diff
@@ -85,6 +85,10 @@ class StorageDistributed final : public IStorage, WithContext
     bool supportsPrewhere() const override { return true; }
     bool supportsSubcolumns() const override { return true; }
     bool supportsDynamicSubcolumns() const override { return true; }
+
+    bool supportsOptimizeRewrite() const { return distributed_settings.enable_ddl_optimize_rewrite; }
```
Comment (alexey-milovidov) [inline src/Storages/StorageDistributed.h:89]: It is a complication, we need to consult if it is worth it.

**Comment by kindred77** [comment]
something wrong with my DEV environment, will submit another pull request for this.

**Pull request closed without merging**